frontend/app/utils/CampaignCode.scala

```diff
@@ -8,9 +8,8 @@ case class CampaignCode(get: String)
 object CampaignCode {
   def fromRequest(implicit request: RequestHeader): Option[CampaignCode] =
     for {
-      cookie <- request.cookies.get("s_sess")
-      decodedCookie = UriEncoding.decodePathSegment(cookie.value, "utf-8")
-      regexMatch <- "campaign=(.+?);".r.findFirstMatchIn(decodedCookie)
-      code <- Option(regexMatch.group(1))
-    } yield CampaignCode(code)
+      cookie <- request.cookies.get("mem_campaign_code")
+    } yield {
+    	CampaignCode(cookie.value)
```

Comment:
`request.cookies.get` returns an `Option[Cookie]` according the [Play docs](https://playframework.com/documentation/2.0/api/scala/play/api/mvc/Cookies.html).

My understanding is that if `cookie` here is `None` then the code in the yield won't get executed, and the whole expression should return `None`?